xapian-core/weight/boolweight.cc

```diff
@@ -86,4 +86,14 @@ BoolWeight::get_maxextra() const
     return 0;
 }
 
+BoolWeight *
+BoolWeight::create_from_parameters(const char * p) const
+{
+    if (*p == '\0')
+	return new Xapian::BoolWeight();
+    if (*p == ' ')
+	throw "No parameters are required for BoolWeight";
+    return new Xapian::BoolWeight();
```

Comment:
`Weight::create` makes sure that `p` either points to a `\0` or a blank space before `create_from_parameters()` method is called with `p`. I can't see how `"1.7"` is a valid parameter string here although `" 1.7" could be a valid string in this case where `p` would be pointing to a blank space in which case an exception is thrown as specified. I might be missing your point; can you please elaborate a bit?

> Also, the exception type should be Xapian::InvalidArgumentError, not const char*.

Yeah, will fix that.